src/build/util.py

```diff
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: MIT
+
+import os
+import pathlib
+import sys
+import tempfile
+
+from typing import Union
+
+import pep517
+
+import build
+import build.env
+
+
+if sys.version_info >= (3, 8):
+    import importlib.metadata as importlib_metadata
+else:
+    import importlib_metadata
+
+
+def _project_wheel_metadata(builder: build.ProjectBuilder) -> 'importlib_metadata.PackageMetadata':
+    with tempfile.TemporaryDirectory() as tmpdir:
+        path = pathlib.Path(builder.metadata_path(tmpdir))
+        # https://github.com/python/importlib_metadata/pull/343
+        return importlib_metadata.PathDistribution(path).metadata  # type: ignore
+
+
+def project_wheel_metadata(
```

Comment:
Add an ``__all__``  with this in it to mark this method as public :+1: (and hides importlib_metadata)

Reply:
I also usually recommend:

```python
def __dir__() -> Tuple[Any, ...]:
    return __all__
```

This fixes tab completion the REPL or IPython in Python 3.7+ to only show public methods.

Reply:
I am gonna do `__all__` to keep in sync with the other modules, and then you can open a PR moving to that form.